docs/csharp/roslyn-sdk/source-generators-overview.md

```diff
@@ -102,51 +84,48 @@ In this guide, you'll explore the creation of a source generator using the <xref
     }
     ```
 
-6. Replace the contents of the `Execute` method, so that it looks like the following:
+    A source generator needs to both implement the <xref:Microsoft.CodeAnalysis.ISourceGenerator?displayProperty=nameWithType> interface, and have the <xref:Microsoft.CodeAnalysis.GeneratorAttribute?displayProperty=nameWithType>. Not all source generators require initialization, and that is the case with this example implemenation &mdash; where <xref:Microsoft.CodeAnalysis.ISourceGenerator.Initialize%2A?displayProperty=nameWithType> is empty.
 
-    ```csharp
-    public void Execute(GeneratorExecutionContext context)
-    {
-        // find the main method
-        var mainMethod = context.Compilation.GetEntryPoint(context.CancellationToken);
-
-        // build up the source code
-        string source = $@"
-    using System;
-
-    namespace {mainMethod.ContainingNamespace.ToDisplayString()}
-    {{
-        public static partial class {mainMethod.ContainingType.Name}
-        {{
-            static partial void HelloFrom(string name)
-            {{
-                Console.WriteLine($""Generator says: Hi from '{{name}}'"");
-            }}
-        }}
-    }}
-    ";
-        // add the source code to the compilation
-        context.AddSource("generatedSource", source);
-    }
-    ```
+1. Replace the contents of the <xref:Microsoft.CodeAnalysis.ISourceGenerator.Execute%2A?displayProperty=nameWithType> method, with the following implementation:
+
+    :::code source="snippets/source-generators/SourceGenerator/HelloSourceGenerator.cs":::
```

Comment:
Missing language
```suggestion
    :::code language="csharp" source="snippets/source-generators/SourceGenerator/HelloSourceGenerator.cs":::
```

Reply:
The language is inferred for well-known types, for example; files with the _.cs_ extension are automatically mapped as C#.